lib/firehose/rack/consumer.rb

```diff
@@ -33,19 +47,30 @@ def self.query_string_subscriptions(env)
           last_sequence = 0 if last_sequence < 0
           {
             channel: chan,
-            message_sequence: last_sequence
+            last_message_sequence: last_sequence
           }
         end
       end
 
       def self.post_subscriptions(request)
         body = request.body.read
-        subs = JSON.parse(body).map do |chan, last_sequence|
-          last_sequence = 0 if last_sequence < 0
-          {
-            channel: chan,
-            message_sequence: last_sequence
-          }
+        subs = JSON.parse(body).map do |chan, val|
+          # Hash is the newer format subscription message that supports
+          # params
+          if val.is_a? Hash
+            {
+              channel: chan,
+              last_message_sequence: val.delete("last_message_sequence"),
+              params: val
+            }
+          # Otherwise the value of the JSON hash is implicitly the message
+          # sequence
+          else
```

Comment:
Is this old format deprecated? If so, let's log a deprecation warning. Then we can remove this conditional in the next major version bump.